renderer/components/UI/Countdown.js

```diff
@@ -39,15 +40,17 @@ class Countdown extends React.Component {
       this.expiresIn = expiresIn
     }
 
-    if (expiresIn >= 0) {
+    // The largest number (ms) supported by setInterval is 2147483647.
+    // If countdown expire date is within this range setup a timer that updates isExpired state on expiry.
+    if (inRange(this.expiresIn, 0, 2147483)) {
```

Comment:
And if it's not in range (>~25 days?) That's a rare case, but if expiry is out of range we could setup a timer that will check in 2147483 secs again and if it's again out of range setup the timer again and so on, until expiry is in range. This probably has a very limited real world application, but would bug free. What do you think?

Reply:
I've done a refactor of this component to convert it to a functional component, simplify it somewhat, and use the `useTimeout` hook to handle updating the isExpired state.

This has resolved the visible bugs that were present and that I was seeking to fix in this PR. Technically, the `useTimeout` hook itself doesn't properly support far future timeouts but in reality this is an edge case, not something that really affects us, and not something that I want to address here.

I've created a separate issue to enhance the `useTimeout` hook to support far future expiry dates. See https://github.com/LN-Zap/zap-desktop/issues/3159